pkg/util/atomic/highwatermark.go

```diff
@@ -0,0 +1,38 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package atomic
+
+import (
+	"sync/atomic"
+)
+
+// HighWaterMark is a thread-safe object for tracking the maximum value seen
+// for some quantity.
+type HighWaterMark int64
```

Comment:
It seems useful, but it has exactly one caller and is a trivial abstraction.

The rename was because it does a Check AND Write.

On Sun, Oct 25, 2015 at 11:15 PM, Harry Zhang <notifications@github.com>
wrote:

> In pkg/util/atomic/highwatermark.go
> <https://github.com/kubernetes/kubernetes/pull/16237#discussion_r42962410>
> :
>
> > +Unless required by applicable law or agreed to in writing, software
> > +distributed under the License is distributed on an "AS IS" BASIS,
> > +WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
> > +See the License for the specific language governing permissions and
> > +limitations under the License.
> > +*/
> > +
> > +package atomic
> > +
> > +import (
> > +	"sync/atomic"
> > +)
> > +
> > +// HighWaterMark is a thread-safe object for tracking the maximum value seen
> > +// for some quantity.
> > +type HighWaterMark int64
>
> Em, I'm not sure ... HighWaterMark seems useful. Should we hide it here?
> BTW, why we rename Check to Update here? Check sounds more meaningful to
> me ...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/16237/files#r42962410>.
>